clipper_admin/clipper_admin/kubernetes/kubernetes_metric_utils.py

```diff
@@ -0,0 +1,81 @@
+import yaml
+import os
+from kubernetes.client import V1ConfigMap
+from contextlib import contextmanager
+from kubernetes.client.rest import ApiException
+import logging
+import json
+from ..version import __version__
+
+_cur_dir = os.path.dirname(os.path.abspath(__file__))
+prom_depolyment_path = os.path.join(_cur_dir, 'prom_depolyment.yml')
+prom_service_path = os.path.join(_cur_dir, 'prom_service.yml')
+prom_configmap_path = os.path.join(_cur_dir, 'prom_configmap.yml')
+frontend_exporter_deployment_path = os.path.join(
+    _cur_dir, 'frontend-exporter-deployment.yaml')
+
+logger = logging.getLogger(__name__)
+
+CLIPPER_FRONTEND_EXPORTER_IMAGE = "clipper/frontend-exporter:{}".format(
+    __version__)
+
+
+@contextmanager
+def _pass_conflicts():
+    try:
+        yield
+    except ApiException as e:
+        body = json.loads(e.body)
+        if body['reason'] == 'AlreadyExists':
+            logger.info("{} already exists, skipping!".format(body['details']))
+            pass
+        else:
+            raise e
+
+
+def _create_prometheus_configmap(_k8s_v1):
+    with open(prom_configmap_path, 'r') as f:
+        data = yaml.load(f)
+
+    with _pass_conflicts():
+        _k8s_v1.create_namespaced_config_map(body=data, namespace='default')
+
+
+def _create_prometheus_depolyment(_k8s_beta):
+    with open(prom_depolyment_path, 'r') as f:
+        data = yaml.load(f)
+
+    with _pass_conflicts():
+        _k8s_beta.create_namespaced_deployment(body=data, namespace='default')
+
+
+def _create_prometheus_service(_k8s_v1):
+    with open(prom_service_path, 'r') as f:
+        data = yaml.load(f)
+
+    with _pass_conflicts():
+        _k8s_v1.create_namespaced_service(body=data, namespace='default')
+
+
+def _create_frontend_exporter_depolyment(_k8s_beta, query_addr):
```

Comment:
What was the reasoning behind the decision to have the frontend exporter in a separate deployment from the query frontend, rather than in the same pod?

Reply:
My initial concern is that: If they are in the same pod, the failure of frontend-exporter will cause continuous restart within the pod. Query-frontend is critical and frontend-exporter is not. 

But if we put them in the same pod, it will definitely make our lives easier when we implement query-frontend scale out. 

I just put the exporter back into the same pod